packages/env-tests/src/shared/attestation.ts

```diff
@@ -0,0 +1,256 @@
+      if (!code) {
+        continue
+      }
+
+      const issuer = await attestations.findMatchingIssuer(
+        identifier,
+        account,
+        code,
+        attestationsToComplete.map((_) => _.issuer)
+      )
+
+      if (!issuer) {
+        continue
+      }
+
+      const isValid = await attestations.validateAttestationCode(identifier, account, issuer, code)
+
+      if (!isValid) {
+        continue
+      }
+
+      return { code, issuer }
+    } catch {
+      continue
+    }
+  }
+
+  return
+}
+
+export async function getPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const phoneNumber = await chooseFromAvailablePhoneNumbers(twilioClient)
+
+  if (phoneNumber !== undefined) {
+    return phoneNumber
+  }
+
+  return createPhoneNumber(twilioClient, addressSid)
+}
+
+async function chooseFromAvailablePhoneNumbers(twilioClient: Twilio) {
+  const availableNumbers = await twilioClient.incomingPhoneNumbers.list()
+  const usableNumber = availableNumbers[0]
+  return usableNumber
+}
+async function createPhoneNumber(twilioClient: Twilio, addressSid: string) {
+  const countryCodes = ['GB', 'US']
+  const countryCode = sample(countryCodes)
+  const context = await twilioClient.availablePhoneNumbers.get(countryCode!)
+  const numbers = await context.mobile.list({ limit: 10 })
+  const usableNumber = numbers[0]
+
+  await twilioClient.incomingPhoneNumbers.create({
+    phoneNumber: usableNumber!.phoneNumber,
+    addressSid,
+    // Just an requestbin.com endpoint to avoid errors
+    smsUrl: 'https://enzyutth0wxme.x.pipedream.net/',
+  })
+
+  return usableNumber
+}
+
+async function fetchLatestMessagesFromToday(client: Twilio, phoneNumber: string, count: number) {
+  return client.messages.list({
+    to: phoneNumber,
+    pageSize: count,
+    // Twilio keeps track of dates in UTC so it could be yesterday too
+    dateSentAfter: moment()
+      .subtract(2, 'day')
+      .toDate(),
+  })
+}
+
+export async function pollForMessagesAndCompleteAttestations(
+  attestations: AttestationsWrapper,
+  client: Twilio,
+  phoneNumber: string,
+  identifier: string,
+  account: Address,
+  attestationsToComplete: ActionableAttestation[],
+  txParams: CeloTransactionParams = {},
+  logger: Logger,
+  timeToPollForTextMessages: number
+) {
+  const startDate = moment()
+  logger.info({ pollingWait: POLLING_WAIT }, 'Poll for the attestation code')
+  while (
+    moment.duration(moment().diff(startDate)).asMinutes() < timeToPollForTextMessages &&
+    attestationsToComplete.length > 0
+  ) {
+    const messages = await fetchLatestMessagesFromToday(client, phoneNumber, 100)
+
+    const res = await findValidCode(
+      attestations,
+      messages.map((_) => _.body),
+      identifier,
+      attestationsToComplete,
+      account
+    )
+
+    if (!res) {
+      await sleep(POLLING_WAIT)
+      continue
+    }
+
+    logger.info(
+      { waitingTime: moment.duration(moment().diff(startDate)).asSeconds() },
```

Comment:
probably not useful but shameless plug for `secondsToDurationString`

Reply:
I think that is great for human output, but this is meant as structured logging :)